zcash_proofs/src/sapling/prover.rs

```diff
@@ -85,19 +85,25 @@ impl SaplingProvingContext {
         // Let's compute the nullifier while we have the position
         let note = Note::from_parts(payment_address, NoteValue::from_raw(value), rseed);
 
-        let nullifier = note.nf(&viewing_key.nk, merkle_path.position());
+        let nullifier = note.nf(
+            &viewing_key.nk,
+            u64::try_from(merkle_path.position())
+                .expect("Sapling note commitment tree position must fit into a u64"),
```

Comment:
I've seen this in three separate places, which is a good indication that we should have a wrapping or extension type.